packages/core/src/sync-store/postgres/store.ts

```diff
@@ -198,9 +209,18 @@ export class PostgresSyncStore implements SyncStore {
                 .values(mergedIntervalRows)
                 .execute();
             }
-          });
-        }),
-      );
+
+            if (existingIntervals.length !== MAX_MERGE_INTERVALS) break;
+
+            if (mergedIntervalRows.length === 0) {
+              // This occurs when there are too many non-mergeable ranges with the same logFilterId. Should be almost impossible.
+              throw new NonRetryableError(
+                `'logFilterIntervals' table for chain '${chainId}' has reached an unrecoverable level of fragmentation. Wiping your database is recommended.`,
```

Comment:
Would omit the 2nd sentence. If this happens, I think ideally the user would report it so we can investigate.